### DISCUSSION TITLE
bibtex error reporting

### DISCUSSION THREAD

**Discussion by NWermes** [WinEdt-Team/WinEdt-Support#33] [1.1 General Help and Discussion]
When I compile winedt (PDFLATEX) I get

PDFLaTeX Compilation Report (Pages: 111)
Errors: 0   Warnings: 9   Bad Boxes: 120

but bibtex errors are not reported. 
Other editors show for example

﻿﻿Latex Result:    0 Errors , 3 Warnings , 103 Bad Boxes , 1010 pages
Bibtex Result:  0 Errors , 0 Warnings

Is it possible to get this information also in WinEdt? Else, what is the easiest way to know whether there are problems
in bibtex references?

**Comment by Alex-Simonic** [comment]
By modifying the macro `%b\Exec\ShowReport.edt` you can display BibTeX Errors and Warnings and you can change the report format to match "what other editors do". If you don't know how to do it let me know and I will consider putting it in the defaults if anyone else is interested...

If you execute BibTeX alone you can already get a report if you enable the option "Show Report" for BibTeX in execution modes (by default it is disabled). If you suspect such problems you should probably do this to get a more detailed error description (produced by BibTeX) that will help you fix the issue...

**Reply by NWermes** [reply]
Thanks for the quick reply.
Unfortunately, I need some more help to modify the  %b\Exec\ShowReport.edt macro as I have not written a macro so far and
failed to find how to edit it. Furthermore, in Execution Modes I do find BibTex in Accessories as "BibTex (?)" with "Show Report" enabled. When I execute "Bibtex" (Ctrl+Shift+B), however, the console window disappears and no results are visible.

**Reply by Alex-Simonic** [reply]
What version and build of WinEdt are you using?

There should be no question mark next to BibTeX and the Pause option in Console attributes should be enabled so the "Window does not disappear". This is how it should look:

![image](https://github.com/WinEdt-Team/WinEdt-Support/assets/157834149/b8a4d397-41b1-4b0c-acd4-a26ae4a78e29)

You can always use Ctrl+Shift+C to manually show and hide WinEdt Console. 

Modifying the macro is probably beyond your skill but I will consider adding BibTeX report for [PDF]TeXify in the defaults if the option is enabled and BibTeX (or Biber) was executed during the compilation...

**Reply by NWermes** [reply]
I am using WinEdt 10.3 Build: 20180507  (v. 10.3) - 64-bit

My Execution Modes Window looks like this
![image](https://github.com/WinEdt-Team/WinEdt-Support/assets/62874643/b34fed1b-a9e8-4071-8b38-3dea920b968e)

Yes, macro programming likely is beyond my current skills. I would appreciate the mentioned add-on.

**Comment by Alex-Simonic** [comment]
If you don't want the console window to disappear Check the Pause / Do not Hide Window (as explained in the previous reply).

If you want (?) to "go away" change the definition from bibtex to bibtex.exe (as indicated in my image) and press OK. This is the default but it will work without .exe as well (thus this is not a problem).

The next version of WinEdt 11 will produce:

```
 PDFTeXify Compilation Report (Pages: 65)
 LaTeX (lualatex.exe) Report:  Errors: 0,  Warnings: 0,  Bad Boxes: 0
 BibTeX (biber.exe) Report:  Errors: 0,  Warnings: 0
```
if you have Show Report enabled for BibTeX. Alas, this will not work in your older version and I will not publish the modified macros as an add-on for older versions (too much trouble)...
